extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -328,9 +328,12 @@ export class JupyterServerInstallation {
 		}
 
 		if (this._installInProgress) {
-			return Promise.reject(msgPendingInstallError);
+			this.apiWrapper.showInfoMessage(msgWaitingForInstall);
+			return this._installCompletion.promise;
```

Comment:
do we need to initialize this._installlCompletion before this time?

Reply:
The assumption is that the Deferred will be initialized after setting installInProgress=true elsewhere.

Reply:
installInProgress kind of functions like a lock in this class, since lots of notebooks can call the installation code at the same time.